fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift

```diff
@@ -0,0 +1,214 @@
+  func simulatorEvent()
+}
+
+public class HumanReadableEventReporter : NSObject, EventReporter {
+  unowned let simulator: FBSimulator
+  let writer: Writer
+  let keywords: [Format.Keywords]
+
+  init(simulator: FBSimulator, writer: Writer, keywords: [Format.Keywords]) {
+    self.simulator = simulator
+    self.writer = writer
+    self.keywords = keywords
+    super.init()
+    self.simulator.userEventSink = self
+  }
+
+  public func containerApplicationDidLaunch(applicationProcess: FBProcessInfo!) {
+
+  }
+
+  public func containerApplicationDidTerminate(applicationProcess: FBProcessInfo!, expected: Bool) {
+
+  }
+
+  public func simulatorDidLaunch(launchdSimProcess: FBProcessInfo!) {
+
+  }
+
+  public func simulatorDidTerminate(launchdSimProcess: FBProcessInfo!, expected: Bool) {
+
+  }
+
+  public func agentDidLaunch(launchConfig: FBAgentLaunchConfiguration!, didStart agentProcess: FBProcessInfo!, stdOut: NSFileHandle!, stdErr: NSFileHandle!) {
+
+  }
+
+  public func agentDidTerminate(agentProcess: FBProcessInfo!, expected: Bool) {
+
+  }
+
+  public func applicationDidLaunch(launchConfig: FBApplicationLaunchConfiguration!, didStart applicationProcess: FBProcessInfo!, stdOut: NSFileHandle!, stdErr: NSFileHandle!) {
+
+  }
+
+  public func applicationDidTerminate(applicationProcess: FBProcessInfo!, expected: Bool) {
+
+  }
+
+  public func diagnosticInformationAvailable(name: String!, process: FBProcessInfo!, value: protocol<NSCoding, NSCopying>!) {
+
+  }
+
+  public func didChangeState(state: FBSimulatorState) {
+    self.report(EventName.StateChange, EventType.Discrete, state.description)
+  }
+
+  public func terminationHandleAvailable(terminationHandle: FBTerminationHandleProtocol!) {
+
+  }
+
+  public func report(eventName: EventName, _ eventType: EventType, _ subject: EventReporterSubject) {
+    self.writer.write("\(self.formattedSimulator):  \(eventName.rawValue) with \(subject.shortDescription)")
+  }
+
+  public func simulatorEvent() {
+    self.writer.write(self.formattedSimulator)
+  }
+
+  private var formattedSimulator: String {
+    get {
+      let tokens: [String] = self.keywords.map { keyword in
+        switch keyword {
+        case .UDID:
+          return simulator.udid
+        case .Name:
+          return simulator.name
+        case .DeviceName:
+          guard let configuration = simulator.configuration else {
+            return "unknown-name"
+          }
+          return configuration.deviceName
+        case .OSVersion:
+          guard let configuration = simulator.configuration else {
+            return "unknown-os"
+          }
+          return configuration.osVersionString
+        case .State:
+          return simulator.stateString
+        case .ProcessIdentifier:
+          guard let process = simulator.launchdSimProcess else {
+            return "no-process"
+          }
+          return process.processIdentifier.description
+        }
+      }
+      return tokens.joinWithSeparator(" ")
```

Comment:
What does it mean? a name "a b" could be interpreted as two tokens? Should we use commas?